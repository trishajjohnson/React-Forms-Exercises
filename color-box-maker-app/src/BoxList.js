import React, { useState } from "react";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";
// import {v4 as uuid} from "uuid";

function BoxList() {
    const [boxes, setBoxes] = useState([]);

    const addBox = box => {
        // let newBox = {...box, id: uuid()};
        setBoxes(boxes => [...boxes, box]);
    };

    const removeBox = id => {
        setBoxes(boxes => boxes.filter(box => box.id !== id));
    };

    const allBoxes = boxes.map(box => (
        <Box 
            key = {box.id}
            id = {box.id}
            backgroundColor = {box.backgroundColor}
            width = {box.width}
            height = {box.height}
            handleRemove = {removeBox}
        />
    ));

    return (
        <div className="Boxlist">
            <NewBoxForm createBox={addBox}/>
            {allBoxes}
        </div>
    );
}

export default BoxList;