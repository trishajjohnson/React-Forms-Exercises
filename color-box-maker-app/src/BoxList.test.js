import React from "react";
import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

// Smoke test for BoxList component
it("should render without crashing", function() {
  render(<BoxList />)
});

// Snapshot test for BoxList component
it("should match snapshot", function() {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

it("should add new box", function() {
    const {queryByText, getByLabelText} = render(<BoxList />);
    expect(queryByText("X")).not.toBeInTheDocument();

    const height = getByLabelText("Box Height");
    const width = getByLabelText("Box Width");
    const backgroundColor = getByLabelText("Box Color");
    const btn = queryByText("Add Box");
    fireEvent.change(height, { target: { value: "5" } });
    fireEvent.change(width, { target: { value: "5" } });
    fireEvent.change(backgroundColor, { target: { value: "red" } });
    fireEvent.click(btn);
    expect(queryByText("X")).toBeInTheDocument();

});

it("should delete box", function() {
    const {queryByText, getByLabelText} = render(<BoxList />);
    const height = getByLabelText("Box Height");
    const width = getByLabelText("Box Width");
    const backgroundColor = getByLabelText("Box Color");
    const btn = queryByText("Add Box");
    fireEvent.change(height, { target: { value: "5" } });
    fireEvent.change(width, { target: { value: "5" } });
    fireEvent.change(backgroundColor, { target: { value: "red" } });
    fireEvent.click(btn);
    expect(queryByText("X")).toBeInTheDocument();

    const deleteBtn = queryByText("X");
    fireEvent.click(deleteBtn);

    expect(queryByText("X")).not.toBeInTheDocument();

});