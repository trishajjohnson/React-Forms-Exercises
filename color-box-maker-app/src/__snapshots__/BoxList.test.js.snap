// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Boxlist"
  >
    <form>
      <label
        for="height"
      >
        Box Height
      </label>
      <input
        id="height"
        name="height"
        type="text"
        value=""
      />
      <label
        for="width"
      >
        Box Width
      </label>
      <input
        id="width"
        name="width"
        type="text"
        value=""
      />
      <label
        for="backgroundColor"
      >
        Box Color
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        type="text"
        value=""
      />
      <button>
        Add Box
      </button>
    </form>
  </div>
</DocumentFragment>
`;
